### ISSUE TITLE
not possible to turn off auto-selection

### ISSUE THREAD

**Issue by coderazzi** [coderazzi/tablefilter-swing#33]
(Original issue 33 created by None on 2013-06-12T13:48:45+00:00)

```
There is no &quot;setAutoSelection&quot; delegate on the TableFilterHeader class and
since FilterHandler is private the only way to prevent auto-selection is to do this
via the global settings singleton. 

The ugly workaround I found is to use custom &quot;RowSorter&quot; that ignores the
&quot;addRowSorterListener&quot; call performed in the &quot;FilterHandler&quot;.
```

Original issue reported on code.google.com by `vojta.janota` on 2013-06-12 13:48:45

**Comment by coderazzi** [comment]
(Original issue 33 last updated on 2013-06-12T13:48:45+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by coderazzi** [comment]
(Original comment by coderazzi on 2013-06-16T16:15:00+00:00)

```
Why don't you better use the global settings? 
Once that you define one table to have the autoselection enabled, the user would likely
expect the same behavior in all the tables with filters, right?
At least, that was my reasoning to avoid defining it in the TableFilterHeader' API,
using instead a global variable. 
It is an easy to implement feature, just not sure whether it provides real value.
```

Original issue reported on code.google.com by `coderazzi` on 2013-06-16 16:15:00

**Comment by coderazzi** [comment]
(Original comment by None on 2013-06-16T17:23:14+00:00)

```
I'm using tablefilter in two different contexts. In one the auto-select makes sense,
in other it does not. The first case is the usual scenario where I want the toolbar
actions to reflect the only existing row. In the other however the auto-selection is
problematic. The table there is used to allow selection of (many) choices out of many.
Auto-selection causes filtered row to be auto-selected (zero column holds a checkbox)
which is not the desired behaviour. The filter should simply narrow the list of available
choices, but there is no reason to assume that the user really wants to toggle given
choice.
```

Original issue reported on code.google.com by `vojta.janota` on 2013-06-16 17:23:14

**Comment by coderazzi** [comment]
(Original comment by coderazzi on 2013-06-17T22:10:47+00:00)

```
(No text was entered with this change)
```

Original issue reported on code.google.com by `coderazzi` on 2013-06-17 22:10:47

**Comment by coderazzi** [comment]
(Original comment by coderazzi on 2013-06-29T19:57:06+00:00)

```
Released version 5.1.1 with this update.
```

Original issue reported on code.google.com by `coderazzi` on 2013-06-29 19:57:06
